ParcelGoApp/views.py

```diff
@@ -322,3 +326,66 @@ def link_package_status(request, tracking_number):
         return render(
             request, "ParcelGoApp/link_package_status.html", {"message": message}
         )
+
+@staff_member_required
+def parcel_assignment(request):
+    current_user = request.user
+    if request.method == "POST":
+        approved_parcel_ids = request.POST.getlist("approved_parcels")
+        selected_courier = request.POST.get("select_courier")
+
+        for parcel_id in approved_parcel_ids:
+            parcel = Parcel.objects.get(id=parcel_id)
```

Comment:
Zamiast pętli for I w jednej linii, z wykorzystaniem update a nie save.  Parcel.objects.filter(id__in=approved_parcel_ids).update(couriers=selected_coutier)